cgosqlite/cgosqlite.go

```diff
@@ -150,6 +170,15 @@ func (db *DB) AutoCheckpoint(n int) error {
 	return errCode(res)
 }
 
+func (db *DB) SetWALHook(f func(dbName string, pages int)) {
+	walHookMu.Lock()
+	handle := C.uintptr_t(len(walHooks))
+	walHooks[handle] = f
+	walHookMu.Unlock()
+
+	C.our_sqlite3_wal_hook(db.db, handle)
```

Comment:
There are two pieces of cleanup to add at some point:

1. On `SetWALHook(nil)` delete the map entry. (minor)
2. On Close, delete the map entry, to avoid leaking closures

TODOs would be fine.

Reply:
Fixed.